From 543b77f30ce0ac263e88800f63761195a558a9eb Mon Sep 17 00:00:00 2001
From: Dmitry Monakhov <dmonakhov@openvz.org>
Date: Sat, 7 Apr 2012 04:22:14 +0000
Subject: [PATCH] expat: Prepare libjingle specific build

- Add HAVE_CONFIG define
- Disable all unnecessary targets in order to reduce build time
---
 Android.mk |   43 +------------------------------------------
 1 files changed, 1 insertions(+), 42 deletions(-)

diff --git a/Android.mk b/Android.mk
index faef5bc..db43793 100644
--- a/Android.mk
+++ b/Android.mk
@@ -8,7 +8,7 @@ common_SRC_FILES := \
 	lib/xmlrole.c \
 	lib/xmltok.c
 
-common_CFLAGS := -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -DHAVE_EXPAT_CONFIG_H
+common_CFLAGS := -Wall -Wmissing-prototypes -Wstrict-prototypes -fexceptions -DHAVE_EXPAT_CONFIG_H -DXML_STATIC
 
 common_C_INCLUDES += \
 	$(LOCAL_PATH)/lib
@@ -18,27 +18,6 @@ common_COPY_HEADERS := \
 	lib/expat.h \
 	lib/expat_external.h
 
-# For the host
-# =====================================================
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(common_SRC_FILES)
-LOCAL_CFLAGS += $(common_CFLAGS)
-LOCAL_C_INCLUDES += $(common_C_INCLUDES)
-
-ifeq ($(HOST_OS),darwin)
-	LOCAL_CFLAGS += -fno-common
-endif
-
-LOCAL_MODULE:= libexpat
-LOCAL_MODULE_TAGS := optional
-LOCAL_COPY_HEADERS_TO := $(common_COPY_HEADERS_TO)
-LOCAL_COPY_HEADERS := $(common_COPY_HEADERS)
-
-include $(BUILD_HOST_STATIC_LIBRARY)
-
-
 # For the device
 # =====================================================
 
@@ -58,23 +37,3 @@ LOCAL_MODULE:= libexpat_static
 LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_STATIC_LIBRARY)
-
-# Device shared library
-include $(CLEAR_VARS)
-
-ifeq ($(TARGET_ARCH),arm)
-LOCAL_NDK_VERSION := 4
-LOCAL_SDK_VERSION := 8
-endif
-
-LOCAL_SRC_FILES := $(common_SRC_FILES)
-LOCAL_CFLAGS += $(common_CFLAGS)
-LOCAL_C_INCLUDES += $(common_C_INCLUDES)
-
-LOCAL_MODULE:= libexpat
-LOCAL_MODULE_TAGS := optional
-LOCAL_COPY_HEADERS_TO := $(common_COPY_HEADERS_TO)
-LOCAL_COPY_HEADERS := $(common_COPY_HEADERS)
-
-include $(BUILD_SHARED_LIBRARY)
-
-- 
1.7.5.4

